saleor/order/models.py

```diff
@@ -263,6 +183,67 @@ def __str__(self):
     def get_total(self):
         return self.unit_price * self.quantity
 
+    @property
+    def quantity_unfulfilled(self):
+        return self.quantity - self.quantity_fulfilled
+
+    @property
+    def quantity_fulfilled(self):
+        lines = FulfillmentLine.objects.filter(
```

Comment:
Isn't the logic here too much for a property? I would expect from a property to perform rather simple logic based on model fields. Here we're performing a database query. I'm afraid that sooner or later this will lead to duplicated queries or someone will call it in a template (performing queries from templates isn't a good practice). Wouldn't it be better to denormalize this value and store it in the database? @patrys What's your view on that?

Reply:
👍 from my end. Executing queries in a property adds a hidden cost to something that looks innocent plus there is no way to optimize it.

As for the logic if `FulfillmentLine` objects for the order are already prefetched then iterating over `lines.all()` and filtering in Python is cheaper (as it won't touch the database) than executing a `filter()`.

Reply:
True, if we did the filtering in Python then such logic would be acceptable for a property in my opinion. I assume that anyone using this will have to ensure proper prefetching.